src/record_mixing.c

```diff
@@ -1,28 +1,1582 @@
 IWRAM_DATA OldMan *gUnknown_03001140;
 IWRAM_DATA struct EasyChatPair *gUnknown_03001144;
-IWRAM_DATA struct DaycareData *gUnknown_03001148;
+IWRAM_DATA struct UnkStruct_80E7B60 *gUnknown_03001148;
 IWRAM_DATA void *gUnknown_0300114C; // gSaveBlock2Ptr->field_64C
 IWRAM_DATA LilycoveLady *gUnknown_03001150;
 IWRAM_DATA void *gUnknown_03001154; // gSaveBlock2Ptr->field_0DC;
 IWRAM_DATA void *gUnknown_03001158; // gSaveBlock2Ptr->field_64C
 IWRAM_DATA u32 gUnknown_0300115C;
-IWRAM_DATA u8 gUnknown_03001160[8];
+IWRAM_DATA u8 gUnknown_03001160;
+IWRAM_DATA u32 filler_03001164;
 IWRAM_DATA u32 gUnknown_03001168[3];
 
+EWRAM_DATA struct UnkStruct_80E7B60 gUnknown_02039F9C = {};
+EWRAM_DATA struct PlayerRecords *gUnknown_0203A014 = NULL;
+EWRAM_DATA struct PlayerRecords *gUnknown_0203A018 = NULL;
+
 // Static ROM declarations
 
+static void sub_80E715C(u8 taskId);
+static void sub_80E7324(u8 taskId);
+static void sub_80E756C(u8 taskId);
+static void sub_80E7630(u8 taskId);
+static void sub_80E77D4(u8 taskId);
+static void *sub_80E77FC(const u16 *asShort);
+static void sub_80E7808(void *records, u16 *a1);
+static u8 sub_80E7810(void);
+static void *sub_80E7820(u8);
+static void sub_80E78C4(OldMan *, size_t, u8);
+static void sub_80E7948(union BattleTowerRecord *, size_t, u8);
+static void sub_80E7A14(LilycoveLady *, size_t, u8);
+static void sub_80E7B2C(const u8 *);
+static void sub_80E7B60(struct UnkStruct_80E7B60 *, size_t, u8, TVShow *);
+static void sub_80E7F68(u16 *item, u8 which);
+static void sub_80E7FF8(u8 taskId);
+/*static*/ void sub_80E8110(struct UnkRecordMixingStruct *arg0, struct UnkRecordMixingStruct *arg1);
+void sub_80E8468(void *, size_t, u8);
+void sub_80E89AC(void *, size_t, u8);
+void sub_80E89F8(void *dest);
+void sub_80E8A54(void *src);
+void sub_80E8AC0(union BattleTowerRecord *);
+void sub_80EAF80(struct SecretBaseRecord *, size_t, u8);
+void sub_812287C(struct EasyChatPair *, size_t, u8);
+void TaskDummy4(union BattleTowerRecord *src);
+
 // .rodata
 
+extern const u8 gUnknown_0858CF8C[];
+extern const u8 gUnknown_0858CF8E[][3];
+extern const u8 gUnknown_0858CF94[][4];
+
+extern const u8 gUnknown_0858CFB8[3][2];
+extern const u8 gUnknown_0858CFBE[3][4];
+
 // .text
+
+void sub_80E6BE8(void)
+{
+    sub_80B37D4(sub_80E715C);
+}
+
+void sub_80E6BF8(void)
+{
+    gUnknown_03001134 = gSaveBlock1Ptr->secretBases;
+    gUnknown_03001138 = gSaveBlock1Ptr->tvShows;
+    gUnknown_0300113C = gSaveBlock1Ptr->pokeNews;
+    gUnknown_03001140 = &gSaveBlock1Ptr->oldMan;
+    gUnknown_03001144 = gSaveBlock1Ptr->easyChatPairs;
+    gUnknown_03001148 = &gUnknown_02039F9C;
+    gUnknown_0300114C = gSaveBlock2Ptr->field_64C;
+    gUnknown_03001150 = &gSaveBlock1Ptr->lilycoveLady;
+    gUnknown_03001154 = gSaveBlock2Ptr->field_0DC;
+    gUnknown_03001158 = gSaveBlock2Ptr->field_64C;
+}
+
+void sub_80E6CA0(struct PlayerRecords *dest)
+{
+    memcpy(dest->secretBases, gUnknown_03001134, sizeof(struct SecretBaseRecord) * 20);
```

Comment:
These pointers should probably be made into arrays of the length that's being multiplied here, then use sizeof(*ptr)